Src/WindowsServer/WindowsServer.Shared.Tests/AzureInstanceMetadataTests.cs

```diff
@@ -0,0 +1,64 @@
+namespace Microsoft.ApplicationInsights.WindowsServer
```

Comment:
AzureMetadataRequestor is the class which does the actual work of querying Azure endpoint, and it looks like it is not tested fully.
Can you unit test it by mocking the endpoint it queries? It should validate scenarios like the endpoint does not respond, responds with non sucess, responds with HUGE payload etc.

Reply:
I stuck to unit tests this far, but I agree we do need to test through this path as well. I'll add some dependency injection feature for this class so I can get around the actual call to the URL (and mock it out). This might take until EOD.

Reply:
https://github.com/Microsoft/ApplicationInsights-dotnet-server/blob/develop/Src/DependencyCollector/NetCore.Tests/DependencyTrackingTelemetryModuleTestNetCore.cs#L191
We have lot of examples to mock any http endpoint. Please take a look at them.

Reply:
It does not look like this comment is addressed?

Reply:
It is, but in the interest of getting this initial (beta) functionality in place for beta1 I created an issue to track it and pushed it to beta2 (plus, there will likely be further functional/integration tests further to the ones you brought up as well). See issue #832.